core/src/main/java/edu/wpi/grip/core/operations/composite/PublishVideoOperation.java

```diff
@@ -167,25 +142,109 @@ public PublishVideoOperation(InputSocket.Factory inputSocketFactory) {
 
   @Override
   public void perform() {
-    if (!connected) {
-      return; // Don't waste any time converting images if there's no dashboard connected
+    final long now = System.nanoTime(); // NOPMD
+
+    if (!cscoreLoaded) {
+      throw new IllegalStateException(
+          "cscore could not be loaded. The image streaming server cannot be started.");
     }
 
-    if (inputSocket.getValue().get().empty()) {
+    opencv_core.Mat input = inputSocket.getValue().get();
+    if (input.empty() || input.isNull()) {
       throw new IllegalArgumentException("Input image must not be empty");
     }
 
-    synchronized (imageLock) {
-      imencode(".jpeg", inputSocket.getValue().get(), imagePointer,
-          new IntPointer(CV_IMWRITE_JPEG_QUALITY, qualitySocket.getValue().get().intValue()));
-      imageLock.notifyAll();
+    copyJavaCvToOpenCvMat(input, publishMat);
+    // Make sure the output resolution is up to date. Might not be needed, depends on cscore updates
+    serverSource.setResolution(input.size().width(), input.size().height());
+    serverSource.putFrame(publishMat);
+    if (lastFrame != -1) {
+      long dt = now - lastFrame;
```

Comment:
I've seen this actually be `0` even though you are using nano seconds. (This is highly unlikely but we saw it in the camera source code in one bug) This will cause a DIV by zero exception.

Reply:
It'll actually be `Double.Infinity` per IEEE spec for floating-point numbers (`1e9 / 0L = Infinity`)

Reply:
`(int) 1e9 / 0L` will throw `java.lang.ArithmeticException: / by zero` because of the cast.

Reply:
In kotlin: `(1e9 / 0L).toInt()` is equal to `2147483647` which is an insanely high frame rate. Is that what you want to set it to?

Reply:
It's technically infinite since the new frame comes at the same time as the previous one. But it doesn't happen here because it's rate limited by the pipeline and the update rate of the pipeline sources. I seriously doubt `dt` will ever be less than a few hundred microseconds.

Reply:
Okay, that's fine then.